### PR TITLE
HTTP3 bugfix

### PR DESCRIPTION
**Pull request by oxsean**
## What is the purpose of the change?
* Heartbeat support under http3
* Fix thread blocking issue when channel reconnection under http3.

## Checklist
- [x] Make sure there is a [GitHub_issue](https://github.com/apache/dubbo/issues) field for the change.
- [x] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
- [x] Write necessary unit-test to verify your logic correction. If the new feature or significant change is committed, please remember to add sample in [dubbo samples](https://github.com/apache/dubbo-samples) project.
- [x] Make sure gitHub actions can pass. [Why the workflow is failing and how to fix it?](../CONTRIBUTING.md)

### CODE REVIEW

**Review by AlbumenJ — APPROVED** [review]

File: dubbo-remoting/dubbo-remoting-http3/src/main/java/org/apache/dubbo/remoting/transport/netty4/NettyHttp3ConnectionClient.java [hunk 1]
```diff
@@ -81,17 +81,15 @@ protected void initChannel(NioDatagramChannel ch) {
...
                     protected void initChannel(QuicChannel ch) {
-                        ch.pipeline()
-                                .addLast(new IdleStateHandler(heartbeat, 0, 0, MILLISECONDS))
-                                .addLast(Constants.CONNECTION_HANDLER_NAME, connectionHandler)
-                                .addLast(new Http3ClientConnectionHandler());
-
+                        ChannelPipeline pipeline = ch.pipeline();
+                        pipeline.addLast(new Http3ClientConnectionHandler());
+                        pipeline.addLast(Constants.CONNECTION_HANDLER_NAME, connectionHandler);
+                        pipelineConfigurator.accept(pipeline);
```
Comment (finefuture) [inline dubbo-remoting/dubbo-remoting-http3/src/main/java/org/apache/dubbo/remoting/transport/netty4/NettyHttp3ConnectionClient.java:94]: Please ensure that it is not null for safety.

File: dubbo-remoting/dubbo-remoting-http3/src/main/java/org/apache/dubbo/remoting/transport/netty4/NettyHttp3Server.java [hunk 2]
```diff
@@ -18,104 +18,76 @@
...
                     .channel(NioDatagramChannel.class)
-                    .handler(codec)
+                    .handler(Http3Helper.configCodec(Http3.newQuicServerCodecBuilder(), getUrl())
+                            .sslContext(Http3SslContexts.buildServerSslContext(getUrl()))
+                            .tokenHandler(InsecureQuicTokenHandler.INSTANCE)
+                            .handler(new ChannelInitializer<QuicChannel>() {
+                                @Override
+                                protected void initChannel(QuicChannel ch) {
+                                    ChannelPipeline pipeline = ch.pipeline();
+                                    pipelineConfigurator.accept(pipeline);
```
Comment (finefuture) [inline dubbo-remoting/dubbo-remoting-http3/src/main/java/org/apache/dubbo/remoting/transport/netty4/NettyHttp3Server.java:88]: Please ensure that it is not null for safety.

File: dubbo-rpc/dubbo-rpc-triple/src/main/java/org/apache/dubbo/rpc/protocol/tri/h3/Http3ClientFrameCodec.java [hunk 3]
```diff
@@ -80,11 +110,38 @@ public void write(ChannelHandlerContext ctx, Object msg, ChannelPromise promise)
                 return;
             }
             ctx.write(new DefaultHttp3DataFrame(frame.content()), promise);
+        } else if (msg instanceof Http2PingFrame) {
+            sendPing((QuicChannel) ctx.channel());
```
Comment (finefuture) [inline dubbo-rpc/dubbo-rpc-triple/src/main/java/org/apache/dubbo/rpc/protocol/tri/h3/Http3ClientFrameCodec.java:114]: I think implementing a pingpong mechanism for http3 would result in clearer code than reusing http2.

File: dubbo-remoting/dubbo-remoting-http3/src/main/java/org/apache/dubbo/remoting/transport/netty4/NettyHttp3ConnectionClient.java (hunk 1 above)
Comment (oxsean) [reply dubbo-remoting/dubbo-remoting-http3/src/main/java/org/apache/dubbo/remoting/transport/netty4/NettyHttp3ConnectionClient.java:94]: Checking in constructor.

File: dubbo-remoting/dubbo-remoting-http3/src/main/java/org/apache/dubbo/remoting/transport/netty4/NettyHttp3Server.java (hunk 2 above)
Comment (oxsean) [reply dubbo-remoting/dubbo-remoting-http3/src/main/java/org/apache/dubbo/remoting/transport/netty4/NettyHttp3Server.java:88]: Checking in constructor.

File: dubbo-rpc/dubbo-rpc-triple/src/main/java/org/apache/dubbo/rpc/protocol/tri/h3/Http3ClientFrameCodec.java (hunk 3 above)
Comment (oxsean) [reply dubbo-rpc/dubbo-rpc-triple/src/main/java/org/apache/dubbo/rpc/protocol/tri/h3/Http3ClientFrameCodec.java:114]: Considering using the control stream for heartbeat and graceful shutdown, but there is no documentation for this, so had to try it myself and used a simple solution for this version.

Comment (finefuture) [reply dubbo-rpc/dubbo-rpc-triple/src/main/java/org/apache/dubbo/rpc/protocol/tri/h3/Http3ClientFrameCodec.java:114]: Mark to be resolved later.

**Review by finefuture — APPROVED** [review]
LGTM

**Pull request merged**